### PR TITLE
Changes for Django 1.10+ compatibility

### PR DESCRIPTION
**Pull request by JordanReiter**
Backwards incompatible changes to Django 1.10 require Middleware classes to either be rewritten or to inherit from django.utils.deprecation.MiddlewareMixin. Code is rewritten to maintain compatibility with previous versions while working for Django 1.10+.

### CODE REVIEW

**Comment by kousu** [comment]
Did you answer [the call](https://github.com/ecometrica/django-multisite/issues/37) or were you already working on this? Thank you. I am reviewing this now, doing due diligence, but it looks nice, tidy, and elegant to me so it should probably be accepted later today.

**Comment by JordanReiter** [comment]
Didn't see the call; I use this library for a site and it didn't work for Django 1.10 so I made the fix.

**Comment by kousu** [comment]
Do you happen to have a copy of the configuration that was breaking? I tried to reproduce #37 and couldn't. I would like to add a regression test for this before accepting the patch.

**Comment by kousu** [comment]
Hey, sorry, I haven't had a chance to look at this this week. I still need to figure out how to regression test it, which seems like it should be easy but was stumping me. If you have a test handy it would expedite the merge. If not, that's okay, but it will be a little while still before this gets up to PyPI unfortunately.

**Comment by JordanReiter** [comment]
The error occurs when you use the new MIDDLEWARE setting instead of the MIDDLEWARE_CLASSES setting. So you will need to create two different testing settings: one for 1.10 and one for <1.10.

Alternatively, I think if you simply add this to your settings file:

    MIDDLEWARE = list(MIDDLEWARE_CLASSES)

Django will automatically use the new-style when testing.

**Comment by kousu** [comment]
Our current [settings.py](https://github.com/ecometrica/django-multisite/blob/98f51a79e15fb668ed7e4e2182cba0141c8a3cdb/multisite/test_settings.py) doesn't load the middleware, and neither do any of our tests right now, which is what I'm trying to fix.

I tried adding
```
MIDDLEWARE = [
    'multisite.middleware.DynamicSiteMiddleware',
]
```
but it didn't do anything; I would have expected it to crash under Django 1.10, but all the tests passed. And worse, if I spell it like
```
MIDDLEWARE = [
    'ultisite.middleware.DynamicSiteMiddleware',
]
```

or
```
MIDDLEWARE_CLASSES = [
    'ultisite.middleware.DynamicSiteMiddleware',
]
```
all the tests pass too. So our current `settings.py` is flat out ignoring middleware.

I am making some time to work on this today, but I don't suppose you have a stripped down copy of your `settings.py` that you could share for me to compare with, could you?

**Comment by kousu** [comment]
I should have said, none of our tests load the middleware via Django's `settings`. They do create instances of it manually, but we need to make sure the tests match the instructions, seeing as the inconsistency is what #37 ran into.

**Comment by JordanReiter** [comment]
I've written some tests which use Client instead of RequestFactory. This allows you to test behavior when both MIDDLEWARE and MIDDLEWARE_CLASSES are set. The test creates a fake view which puts the current site in the context, then the test ensures that the expected site record matches the given host.

**Comment by kousu** [comment]
Oh awesome @JordanReiter ! Thanks for doing writing those.

So, one problem:
```
(.venv-python2) gauchier:django-multisite nguenthe$ python setup.py test
running test
Searching for tldextract>=1.2
Best match: tldextract 2.0.2
Processing tldextract-2.0.2-py2.7.egg

Using /Users/nguenthe/django-multisite/.eggs/tldextract-2.0.2-py2.7.egg
Searching for requests-file>=1.4
Best match: requests-file 1.4.1
Processing requests_file-1.4.1-py2.7.egg

Using /Users/nguenthe/django-multisite/.eggs/requests_file-1.4.1-py2.7.egg
Searching for requests>=2.1.0
Best match: requests 2.13.0
Processing requests-2.13.0-py2.7.egg

Using /Users/nguenthe/django-multisite/.eggs/requests-2.13.0-py2.7.egg
Searching for idna
Best match: idna 2.2
Processing idna-2.2-py2.7.egg

Using /Users/nguenthe/django-multisite/.eggs/idna-2.2-py2.7.egg
running egg_info
writing requirements to django_multisite.egg-info/requires.txt
writing django_multisite.egg-info/PKG-INFO
writing top-level names to django_multisite.egg-info/top_level.txt
writing dependency_links to django_multisite.egg-info/dependency_links.txt
reading manifest file 'django_multisite.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'django_multisite.egg-info/SOURCES.txt'
running build_ext
Traceback (most recent call last):
  File "setup.py", line 44, in <module>
    'Topic :: Utilities'],
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/setuptools/command/test.py", line 211, in run
    self.run_tests()
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/setuptools/command/test.py", line 234, in run_tests
    **exit_kwarg
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/main.py", line 149, in parseArgs
    self.createTests()
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/main.py", line 158, in createTests
    self.module)
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "/Users/nguenthe/django-multisite/multisite/tests.py", line 930, in <module>
    url(r'^debug/$', debug)
NameError: name 'url' is not defined
```

another is that there are tests for the middleware: `multisite.tests.{DynamicSiteMiddlewareTest,DynamicSiteMiddlewareFallbackTest,DynamicSiteMiddlewareSettingsTest}` already which I would prefer these tests be merged in with.

File: multisite/tests.py [hunk 1]
```diff
@@ -919,6 +919,72 @@ def test_subdomain_depth(self):
...
+    ctx = {'current_site': Site.objects.get_current()}
+    return HttpResponse(tt.render(RequestContext(request, ctx)))
+
+
+urlpatterns = [
+    url(r'^debug/$', debug)
+]
+
+
+@skipIf(django.VERSION < (1, 7), "Only works for Django 1.7+")
```
Comment (kousu) [inline multisite/tests.py]: This should be `django.VERSION < (1.8)`; as written it's saying "django<=1.6".

Comment (kousu) [inline multisite/tests.py]: Oh sorry, I misinterpreted. I forgot this says "skipIf".

**Comment by kousu** [comment]
I have adapted your tests and suggestions to merge them inline. I've added one new test, `multisite.tests.DynamicSiteMiddlewareTest.test_integration()` which uses `self.client.get()` instead of instantiating the middleware manually.

After beating it with enough wooden spoons, we have it working under 1.9:
```
(.venv-python2) gauchier:django-multisite nguenthe$ pip install "django<1.10"
Collecting django<1.10
  Using cached Django-1.9.12-py2.py3-none-any.whl
Installing collected packages: django
  Found existing installation: Django 1.10.5
    Uninstalling Django-1.10.5:
      Successfully uninstalled Django-1.10.5
Successfully installed django-1.9.12
(.venv-python2) gauchier:django-multisite nguenthe$ 
(.venv-python2) gauchier:django-multisite nguenthe$ python setup.py test
running test
running egg_info
writing requirements to django_multisite.egg-info/requires.txt
writing django_multisite.egg-info/PKG-INFO
writing top-level names to django_multisite.egg-info/top_level.txt
writing dependency_links to django_multisite.egg-info/dependency_links.txt
reading manifest file 'django_multisite.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'INSTALL.txt'
writing manifest file 'django_multisite.egg-info/SOURCES.txt'
running build_ext
Creating test database for alias 'default'...
test_create (multisite.tests.AliasTest) ... ok
test_expand_netloc (multisite.tests.AliasTest) ... ok
test_hooks (multisite.tests.AliasTest) ... ok
test_managers (multisite.tests.AliasTest) ... ok
test_repr (multisite.tests.AliasTest) ... ok
test_resolve (multisite.tests.AliasTest) ... ok
test_sync (multisite.tests.AliasTest) ... ok
test_sync_all (multisite.tests.AliasTest) ... ok
test_sync_blank_domain (multisite.tests.AliasTest) ... ok
test_sync_many (multisite.tests.AliasTest) ... ok
test_sync_missing (multisite.tests.AliasTest) ... ok
test_site_domain_changed (multisite.tests.CacheTest) ... ok
test_404 (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_class_view (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_invalid (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_string_class (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_testserver (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_invalid_settings (multisite.tests.DynamicSiteMiddlewareSettingsTest) ... ok
test_case_sensitivity (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_change_domain (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_integration (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_invalid_host (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_invalid_hostport (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_no_redirect (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_no_sites (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_redirect (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_unknown_host (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_unknown_hostport (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_valid_domain (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_valid_domain_port (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_change_site (multisite.tests.SiteCacheTest) ... ok
test_create_site (multisite.tests.SiteCacheTest) ... ok
test_default_key_prefix (multisite.tests.SiteCacheTest) ... ok
test_delete_site (multisite.tests.SiteCacheTest) ... ok
test_get_current (multisite.tests.SiteCacheTest) ... ok
test_multisite_key_prefix (multisite.tests.SiteCacheTest) ... ok
test_multisite_key_prefix_takes_priority_over_default (multisite.tests.SiteCacheTest) ... ok
test_get_current_site (multisite.tests.TestContribSite) ... ok
test_effective_tld (multisite.tests.TestCookieDomainMiddleware) ... ok
test_init (multisite.tests.TestCookieDomainMiddleware) ... ok
test_ip_address (multisite.tests.TestCookieDomainMiddleware) ... ok
test_localpath (multisite.tests.TestCookieDomainMiddleware) ... ok
test_matched_cookies (multisite.tests.TestCookieDomainMiddleware) ... ok
test_no_matched_cookies (multisite.tests.TestCookieDomainMiddleware) ... ok
test_simple_tld (multisite.tests.TestCookieDomainMiddleware) ... ok
test_subdomain_depth (multisite.tests.TestCookieDomainMiddleware) ... ok
test_deferred_site (multisite.tests.TestSiteDomain) ... ok
test_init (multisite.tests.TestSiteDomain) ... ok
test_compare_default_site_id (multisite.tests.TestSiteID) ... ok
test_compare_differing_types (multisite.tests.TestSiteID) ... ok
test_compare_site_ids (multisite.tests.TestSiteID) ... ok
test_context_manager (multisite.tests.TestSiteID) ... ok
test_hash (multisite.tests.TestSiteID) ... ok
test_invalid_default (multisite.tests.TestSiteID) ... ok
test_set (multisite.tests.TestSiteID) ... ok
test_str_repr (multisite.tests.TestSiteID) ... ok
test_default_value (multisite.tests.TestSiteIDHook) ... ok
test_deprecation_warning (multisite.tests.TestSiteIDHook) ... ok
Destroying test database for alias 'default'...

----------------------------------------------------------------------
Ran 58 tests in 0.593s

OK
```

and if I try under 1.10, it reproduces the error seen in #37:
```
(.venv-python2) gauchier:django-multisite nguenthe$ git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
(.venv-python2) gauchier:django-multisite nguenthe$ pip install "django>=1.10"
Collecting django>=1.10
  Using cached Django-1.10.5-py2.py3-none-any.whl
Installing collected packages: django
  Found existing installation: Django 1.9.12
    Uninstalling Django-1.9.12:
      Successfully uninstalled Django-1.9.12
Successfully installed django-1.10.5
(.venv-python2) gauchier:django-multisite nguenthe$ python setup.py test
running test
running egg_info
writing requirements to django_multisite.egg-info/requires.txt
writing django_multisite.egg-info/PKG-INFO
writing top-level names to django_multisite.egg-info/top_level.txt
writing dependency_links to django_multisite.egg-info/dependency_links.txt
reading manifest file 'django_multisite.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'INSTALL.txt'
writing manifest file 'django_multisite.egg-info/SOURCES.txt'
running build_ext
/Users/nguenthe/django-multisite/multisite/hacks.py:20: RuntimeWarning: 'default' cache is LocMemCache, which may cause stale caches.
  site_cache = SiteCache()
Creating test database for alias 'default'...
test_create (multisite.tests.AliasTest) ... ok
test_expand_netloc (multisite.tests.AliasTest) ... ok
test_hooks (multisite.tests.AliasTest) ... ok
test_managers (multisite.tests.AliasTest) ... ok
test_repr (multisite.tests.AliasTest) ... ok
test_resolve (multisite.tests.AliasTest) ... ok
test_sync (multisite.tests.AliasTest) ... ok
test_sync_all (multisite.tests.AliasTest) ... ok
test_sync_blank_domain (multisite.tests.AliasTest) ... ok
test_sync_many (multisite.tests.AliasTest) ... ok
test_sync_missing (multisite.tests.AliasTest) ... ok
test_site_domain_changed (multisite.tests.CacheTest) ... ok
test_404 (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_class_view (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_invalid (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_string_class (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_testserver (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_invalid_settings (multisite.tests.DynamicSiteMiddlewareSettingsTest) ... ok
test_case_sensitivity (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_change_domain (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_integration (multisite.tests.DynamicSiteMiddlewareTest) ... ERROR
test_invalid_host (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_invalid_hostport (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_no_redirect (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_no_sites (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_redirect (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_unknown_host (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_unknown_hostport (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_valid_domain (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_valid_domain_port (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_change_site (multisite.tests.SiteCacheTest) ... ok
test_create_site (multisite.tests.SiteCacheTest) ... ok
test_default_key_prefix (multisite.tests.SiteCacheTest) ... ok
test_delete_site (multisite.tests.SiteCacheTest) ... ok
test_get_current (multisite.tests.SiteCacheTest) ... ok
test_multisite_key_prefix (multisite.tests.SiteCacheTest) ... ok
test_multisite_key_prefix_takes_priority_over_default (multisite.tests.SiteCacheTest) ... ok
test_get_current_site (multisite.tests.TestContribSite) ... ok
test_effective_tld (multisite.tests.TestCookieDomainMiddleware) ... ok
test_init (multisite.tests.TestCookieDomainMiddleware) ... ok
test_ip_address (multisite.tests.TestCookieDomainMiddleware) ... ok
test_localpath (multisite.tests.TestCookieDomainMiddleware) ... ok
test_matched_cookies (multisite.tests.TestCookieDomainMiddleware) ... ok
test_no_matched_cookies (multisite.tests.TestCookieDomainMiddleware) ... ok
test_simple_tld (multisite.tests.TestCookieDomainMiddleware) ... ok
test_subdomain_depth (multisite.tests.TestCookieDomainMiddleware) ... ok
test_deferred_site (multisite.tests.TestSiteDomain) ... ok
test_init (multisite.tests.TestSiteDomain) ... ok
test_compare_default_site_id (multisite.tests.TestSiteID) ... ok
test_compare_differing_types (multisite.tests.TestSiteID) ... ok
test_compare_site_ids (multisite.tests.TestSiteID) ... ok
test_context_manager (multisite.tests.TestSiteID) ... ok
test_hash (multisite.tests.TestSiteID) ... ok
test_invalid_default (multisite.tests.TestSiteID) ... ok
test_set (multisite.tests.TestSiteID) ... ok
test_str_repr (multisite.tests.TestSiteID) ... ok
test_default_value (multisite.tests.TestSiteIDHook) ... ok
test_deprecation_warning (multisite.tests.TestSiteIDHook) ... ok
Destroying test database for alias 'default'...

======================================================================
ERROR: test_integration (multisite.tests.DynamicSiteMiddlewareTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nguenthe/django-multisite/multisite/tests.py", line 227, in test_integration
    resp = self.client.get('/domain/', HTTP_HOST=self.host)
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/django/test/client.py", line 529, in get
    **extra)
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/django/test/client.py", line 333, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/django/test/client.py", line 409, in generic
    return self.request(**r)
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/django/test/client.py", line 476, in request
    response = self.handler(environ)
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/django/test/client.py", line 129, in __call__
    self.load_middleware()
  File "/Users/nguenthe/django-multisite/.venv-python2/lib/python2.7/site-packages/django/core/handlers/base.py", line 82, in load_middleware
    mw_instance = middleware(handler)
TypeError: __init__() takes exactly 1 argument (2 given)

----------------------------------------------------------------------
Ran 58 tests in 0.589s

FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=58 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=58 errors=1 failures=0>
```

Finally! Hooray!

And if I pull in your initial pull request:
```
(.venv-python2) gauchier:django-multisite nguenthe$ git cherry-pick 43e656326ffd6b8b353ce12b2dd0143504684333
[master da3870c] Changes for Django 1.10+ compatibility
 Author: Jordan Reiter <jordanreiter@gmail.com>
 Date: Mon Jan 23 10:21:43 2017 -0500
 1 file changed, 9 insertions(+), 2 deletions(-)
(.venv-python2) gauchier:django-multisite nguenthe$ git log -p
commit da3870c9d065245a99bf2c26c755e9e6e13521c1
Author: Jordan Reiter <jordanreiter@gmail.com>
Date:   Mon Jan 23 10:21:43 2017 -0500

    Changes for Django 1.10+ compatibility
    
    Backwards incompatible changes to Django 1.10 require Middleware classes to either be rewritten or to inherit from django.utils.deprecation.MiddlewareMixin. Code is rewritten to maintain compatibility with previous versions while working for Django 1.10+.

diff --git a/multisite/middleware.py b/multisite/middleware.py
index e0b379d..fce6feb 100644
--- a/multisite/middleware.py
+++ b/multisite/middleware.py
@@ -23,6 +23,12 @@ else:
     def get_cache(cache_alias):
         return caches[cache_alias]
 
+try:
+    # Django > 1.10 uses MiddlewareMixin
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:
+    MiddlewareMixin = object
+
 from django.core.exceptions import ImproperlyConfigured
 from django.core.urlresolvers import get_callable
 from django.db.models.signals import pre_save, post_delete, post_init
@@ -38,8 +44,9 @@ except ImportError:
 from .models import Alias
 
 
-class DynamicSiteMiddleware(object):
-    def __init__(self):
+class DynamicSiteMiddleware(MiddlewareMixin):
+    def __init__(self, *args, **kwargs):
+        super(DynamicSiteMiddleware, self).__init__(*args, **kwargs)
         if not hasattr(settings.SITE_ID, 'set'):
             raise TypeError('Invalid type for settings.SITE_ID: %s' %
                             type(settings.SITE_ID).__name__)

(.venv-python2) gauchier:django-multisite nguenthe$ python setup.py test
running test
running egg_info
writing requirements to django_multisite.egg-info/requires.txt
writing django_multisite.egg-info/PKG-INFO
writing top-level names to django_multisite.egg-info/top_level.txt
writing dependency_links to django_multisite.egg-info/dependency_links.txt
reading manifest file 'django_multisite.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'INSTALL.txt'
writing manifest file 'django_multisite.egg-info/SOURCES.txt'
running build_ext
/Users/nguenthe/django-multisite/multisite/hacks.py:20: RuntimeWarning: 'default' cache is LocMemCache, which may cause stale caches.
  site_cache = SiteCache()
Creating test database for alias 'default'...
test_create (multisite.tests.AliasTest) ... ok
test_expand_netloc (multisite.tests.AliasTest) ... ok
test_hooks (multisite.tests.AliasTest) ... ok
test_managers (multisite.tests.AliasTest) ... ok
test_repr (multisite.tests.AliasTest) ... ok
test_resolve (multisite.tests.AliasTest) ... ok
test_sync (multisite.tests.AliasTest) ... ok
test_sync_all (multisite.tests.AliasTest) ... ok
test_sync_blank_domain (multisite.tests.AliasTest) ... ok
test_sync_many (multisite.tests.AliasTest) ... ok
test_sync_missing (multisite.tests.AliasTest) ... ok
test_site_domain_changed (multisite.tests.CacheTest) ... ok
test_404 (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_class_view (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_invalid (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_string_class (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_testserver (multisite.tests.DynamicSiteMiddlewareFallbackTest) ... ok
test_invalid_settings (multisite.tests.DynamicSiteMiddlewareSettingsTest) ... ok
test_case_sensitivity (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_change_domain (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_integration (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_invalid_host (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_invalid_hostport (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_no_redirect (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_no_sites (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_redirect (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_unknown_host (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_unknown_hostport (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_valid_domain (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_valid_domain_port (multisite.tests.DynamicSiteMiddlewareTest) ... ok
test_change_site (multisite.tests.SiteCacheTest) ... ok
test_create_site (multisite.tests.SiteCacheTest) ... ok
test_default_key_prefix (multisite.tests.SiteCacheTest) ... ok
test_delete_site (multisite.tests.SiteCacheTest) ... ok
test_get_current (multisite.tests.SiteCacheTest) ... ok
test_multisite_key_prefix (multisite.tests.SiteCacheTest) ... ok
test_multisite_key_prefix_takes_priority_over_default (multisite.tests.SiteCacheTest) ... ok
test_get_current_site (multisite.tests.TestContribSite) ... ok
test_effective_tld (multisite.tests.TestCookieDomainMiddleware) ... ok
test_init (multisite.tests.TestCookieDomainMiddleware) ... ok
test_ip_address (multisite.tests.TestCookieDomainMiddleware) ... ok
test_localpath (multisite.tests.TestCookieDomainMiddleware) ... ok
test_matched_cookies (multisite.tests.TestCookieDomainMiddleware) ... ok
test_no_matched_cookies (multisite.tests.TestCookieDomainMiddleware) ... ok
test_simple_tld (multisite.tests.TestCookieDomainMiddleware) ... ok
test_subdomain_depth (multisite.tests.TestCookieDomainMiddleware) ... ok
test_deferred_site (multisite.tests.TestSiteDomain) ... ok
test_init (multisite.tests.TestSiteDomain) ... ok
test_compare_default_site_id (multisite.tests.TestSiteID) ... ok
test_compare_differing_types (multisite.tests.TestSiteID) ... ok
test_compare_site_ids (multisite.tests.TestSiteID) ... ok
test_context_manager (multisite.tests.TestSiteID) ... ok
test_hash (multisite.tests.TestSiteID) ... ok
test_invalid_default (multisite.tests.TestSiteID) ... ok
test_set (multisite.tests.TestSiteID) ... ok
test_str_repr (multisite.tests.TestSiteID) ... ok
test_default_value (multisite.tests.TestSiteIDHook) ... ok
test_deprecation_warning (multisite.tests.TestSiteIDHook) ... ok
Destroying test database for alias 'default'...

----------------------------------------------------------------------
Ran 58 tests in 0.604s

OK
```

It works again! nice!

I'm sorry it took two weeks to get this committed. It's such a small change, but the tests really needed some love before I felt safe adopting it. I'll try to get this put up to PyPI tomorrow.

**Pull request closed without merging**